<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GoogleSheetsModifier</class>
 <widget class="QWidget" name="GoogleSheetsModifier">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>861</width>
    <height>871</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Google Sheets communicator</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,0,0,0,0,0">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_9">
     <property name="leftMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="title">
        <string>API Key (Read only)</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_6">
        <item>
         <widget class="QRadioButton" name="radioButtonAPI_key">
          <property name="text">
           <string>API Key method</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_API_KEY">
          <property name="title">
           <string>API key</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="3,1,1">
             <property name="sizeConstraint">
              <enum>QLayout::SetNoConstraint</enum>
             </property>
             <item>
              <widget class="QLineEdit" name="lineAPI_key_filename">
               <property name="placeholderText">
                <string>File name</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="Button_API_key_file_choose">
               <property name="text">
                <string>Choose File</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="Button_API_key_file_load">
               <property name="text">
                <string>Load</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Key</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="line_API_Key">
               <property name="readOnly">
                <bool>true</bool>
               </property>
               <property name="placeholderText">
                <string>API key</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>OAuth2 (Read/Write)</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <item>
         <widget class="QRadioButton" name="radioButton_OAuth2">
          <property name="text">
           <string>OAuth2 method</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_OAuth2">
          <property name="title">
           <string>Authification</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_10">
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLineEdit" name="lineEdit_OAuth_filename">
               <property name="placeholderText">
                <string>File name</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="Button_OAuth_choosefile">
               <property name="text">
                <string>Choose File</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="Button_OAuth_LoadFile">
               <property name="text">
                <string>Load</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QLineEdit" name="lineClientID">
               <property name="placeholderText">
                <string>Client ID</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <property name="bottomMargin">
              <number>0</number>
             </property>
             <item>
              <widget class="QLineEdit" name="lineClientSecret">
               <property name="placeholderText">
                <string>Client Secret</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="ButtonAuth">
             <property name="text">
              <string>Authificate</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_9">
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Sheet name</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineSheetName">
         <property name="placeholderText">
          <string>Can't be empty</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Spread Sheet ID</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="lineSpreadSheetID">
         <property name="placeholderText">
          <string>Can't be empty</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QProgressBar" name="progressBar">
     <property name="value">
      <number>100</number>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QPushButton" name="ButtonRead">
       <property name="text">
        <string>Read</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonWrite">
       <property name="text">
        <string>Write</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="Button_ReadSheetIDs">
       <property name="text">
        <string>Read Sheet ID's</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="checkBox_Selected_cells_work">
       <property name="text">
        <string>Read/Write selected cells</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="checkBox_readWholeTable">
       <property name="text">
        <string>Read whole table</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_write_option">
       <property name="title">
        <string/>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_11">
        <item>
         <widget class="QRadioButton" name="radioButton_option_rewrite">
          <property name="text">
           <string>Rewrite</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="radioButton_option_append">
          <property name="text">
           <string>Append</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonSave">
       <property name="text">
        <string>Save to file</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonLoad">
       <property name="text">
        <string>Load from file</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QPushButton" name="ButtonAddRow">
       <property name="text">
        <string>Row++</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonRemoveRow">
       <property name="text">
        <string>Row--</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonAddColumn">
       <property name="text">
        <string>Column++</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="ButtonRemoveColumn">
       <property name="text">
        <string>Column--</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QCheckBox" name="checkBox_FlashChanges">
       <property name="text">
        <string>Flash changes</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="Button_Save_settings">
       <property name="text">
        <string>Save settings</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="Button_LoadSettings">
       <property name="text">
        <string>Load settings</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_7" stretch="4,1">
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QTableView" name="tableGoogleSheets"/>
     </item>
     <item>
      <widget class="QTextEdit" name="textEdit">
       <property name="placeholderText">
        <string>application log...</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
